kuka_rsi_hw_interface/include/xml_handler/xml_element.hpp

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 Komáromi Sándor
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef  XML__XML_ELEMENT_H_
+#define  XML__XML_ELEMENT_H_
+
+#include <map>
+#include <vector>
+#include <string>
+#include <cstring>
+#include <cstdlib>
+#include <iostream>
+#include <functional>
+
+#include "xml_param.hpp"
+
+namespace xml
+{
+inline bool operator<(const XMLString & a, const std::string & b)
+{
+  return strncmp(a.data_ptr_, b.c_str(), a.length_) < 0;
+}
+
+inline bool operator<(const std::string & b, const XMLString & a)
+{
+  return strncmp(b.c_str(), a.data_ptr_, a.length_) < 0;
+}
+
+class XMLElement
+{
+private:
+  static XMLString castXMLString(char * & str_ptr);
+
+  std::map<std::string, XMLParam, std::less<>> params_;
+  std::vector<XMLElement> childs_;
+  std::string name_;
+
+public:
+  XMLElement(const std::string & name)
+  : name_(name) {}
+  XMLElement() = default;
+  ~XMLElement() = default;
+
+  inline std::string GetName() const {return name_;}
+  inline void SetName(const std::string & name) {name_ = name;}
+  inline std::vector<XMLElement> GetChilds() const {return childs_;}
```

Comment:
the first one is const, so when it can be called the const getter of the data structures in rsi command handler

Reply:
keep in mind that the function returning the vector by value might allocate the vector's elements dynamically, since it has to be copied once. So I'd remove that function.

Reply:
I think you should return const &, as @VX792 has suggested